First Pass of changes

cccs-gm opened the pull request:
- removes the -g flag
- default behavior is to validate the rule as it is without generating id, fingerprint, version, first_imported, or last_modified if not already present.
- id, fingerprint, version, first_imported, or last_modified are auto generated, if not already present, when the -i or -c flags are used
- changed the help message

cccs-gm committed:
First Pass of changes

- removes the -g flag
- default behavior is to validate the rule as it is without generating id, fingerprint, version, first_imported, or last_modified if not already present.
- id, fingerprint, version, first_imported, or last_modified are auto generated, if not already present, when the -i or -c flags are used
- changed the help message

cccs-gm commented:
There is some confusion about the functionality of the script.

Current functionality:
•	When the script is run without the -i or -c flags it returns valid if the YARA rule would be valid with the -i or -c flags. This results in the script returning valid in the case that the id, fingerprint, version, first_imported or last_modified are missing as they can be generated.
•	When it is run with the -i or -c flags the script validates all of the present metadata tags generates any of the missing metadata tags from the id, fingerprint, version, first_imported or last_modified list.

This results in the confusing situation where a rule will return as valid if it has everything but is missing one or more of id, fingerprint, version, first_imported or last_modified.

Proposed functionality:
•	When the script is run without the -i or -c flags it returns valid only if the YARA rule has all of the required metadata with valid values. There are appropriate error messages for the metadata that can be generated and that using the -i or -c flags will generate those metadata. 
•	When it is run with the -i or -c flags the script validates all of the present metadata tags generates any of the missing metadata tags from the id, fingerprint, version, first_imported or last_modified list.

CCCS-MA commented:
Looks good. Will approve once the read me is updated.

CCCS-MA reviewed (changes requested):
Looks good. Will approve once the read me is updated.

cccs-gm committed:
Updated the README.md and yara_validator_cli.py

- Fixed a type in the new contents of the -h flag
- Updated the README.md to include the new help information from the updated yara_validator_cli.py
